domain/operation/state/task.go

```diff
@@ -47,6 +47,97 @@ func (s *State) GetTask(ctx context.Context, taskID string) (operation.Task, *st
 	return result, outputPath, nil
 }
 
+// StartTask sets the task start time and updates the status to running.
+// Returns [operationerrors.TaskNotFound] if the task does not exist,
+// and [operationerrors.TaskNotPending] if the task is not pending.
```

Comment:
We usually format the returned errors similar to: https://github.com/juju/juju/blob/5e8f5af2c9f1b33b3f05134ef593a07826acca63/domain/relation/state/relation.go#L131-L133